pkg/search/postgres/common.go

```diff
@@ -437,6 +438,10 @@ func entriesFromQueries(
 }
 
 func compileQueryToPostgres(schema *walker.Schema, q *v1.Query, queryFields map[string]searchFieldMetadata, nowForQuery time.Time) (*pgsearch.QueryEntry, error) {
+	if err := validateDerivedFieldDataType(queryFields); err != nil {
```

Comment:
My knowledge of this area is admittedly not deep, but this seems pretty late and broad in the query generation to flag this.  

So this change along with https://github.com/stackrox/stackrox/blob/27a956b5603eff96e2dd517bf3f42d7cec8627ce/pkg/search/postgres/select.go#L140 seem to indicate we don't support select on these at all.  Should we support those?  Do we have a ticket for that we could use as a todo reference?  Should we ensure these are not options when building the query altogether and/or skip them when building the derived fields?

Reply:
The check you pointed out is for SQF, whereas this PR introduces check so that the non-SQF code path, which is what we use everywhere, also hits this path. I believe when implementing SQF, I uncovered the gap in the original implementation but adding the check slipped somehow.

 Should we support those? Do we have a ticket for that we could use as a todo reference? Should we ensure these are not options when building the query altogether and/or skip them when building the derived fields?

Reply:
>  Should we support those? Do we have a ticket for that we could use as a todo reference? 
I believe that the data modeling for such fields need to change. For example, `map` should not be used. Instead, a `repeated message` of `key` and `value` fields should be preferred. The data modeling needs could benefit from another round of TLC.

> Should we ensure these are not options when building the query altogether and/or skip them when building the derived fields?

I believe a big fat note would in the `options.go` file would help.

Reply:
Noted.  I guess with the maps in particular we need to think hard about how we use them.  Since we store them as just jsonb they are easy to deal with when we pull the data.  But as this problem illustrates it becomes cumbersome to have re-usable code that utilizes SQL functions or even attempts to query.

Reply:
With the map being unordered, it is not an ideal structure, even for API usage.